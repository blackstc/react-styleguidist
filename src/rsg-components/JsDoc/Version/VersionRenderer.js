import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';
import Markdown from 'rsg-components/Markdown';

export const styles = () => ({
	versionBlock: {
		marginBottom: 6,
		fontSize: 'inherit',
	},
});

export function VersionRenderer({ classes, tags }) {
	if (!tags || !tags.version) {
		return null;
	}

	return (
		<div className={classes.versionBlock}>
			<Markdown text={`Version: ${tags.version[0].description}`} inline />
		</div>
	);
}
VersionRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	tags: PropTypes.shape({
		version: PropTypes.array,
	}).isRequired,
};
export default Styled(styles)(VersionRenderer);
