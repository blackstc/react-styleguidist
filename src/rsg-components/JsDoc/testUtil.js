import React from 'react';

export function createSimpleJsDocTagTest(Renderer, tag) {
	return describe(Renderer.displayName, () => {
		const oneTag = {
			[tag]: [{
				title: tag,
				description: 'Test *description*',
			}],
		};

		const twoTags = {
			[tag]: [{
				title: tag,
				description: 'Test description',
			}, {
				title: tag,
				description: 'I should not be visible',
			}],
		};

		it('renderer should render', () => {
			const actual = render(
				<Renderer classes={{}} tags={oneTag} />
			);

			expect(actual).toMatchSnapshot();
		});

		it(`renderer should only render one ${tag} tag`, () => {
			const actual = render(
				<Renderer classes={{}} tags={twoTags} />
			);

			expect(actual.text().includes('I should not be visible')).toBe(false);
		});


		it('renderer should not render for empty tags', () => {
			const actual = shallow(
				<Renderer classes={{}} tags={{}} />
			);

			expect(actual.node).toBe(null);
		});
	});
}
