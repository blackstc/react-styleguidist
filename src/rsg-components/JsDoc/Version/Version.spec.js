import { VersionRenderer } from './VersionRenderer';
import { createSimpleJsDocTagTest } from '../testUtil';

createSimpleJsDocTagTest(VersionRenderer, 'version');
