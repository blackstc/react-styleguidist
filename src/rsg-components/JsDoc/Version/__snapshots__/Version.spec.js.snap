// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render 1`] = `
<div>
  <span
    class="base-0-0"
  >
    Version: Test 
    <em
      class="em-0-16"
    >
      description
    </em>
  </span>
</div>
`;
